WordPress/wordpress-playground
# WP-692 Add runSql Blueprint step

## What is this PR doing?

This PR addresses https://github.com/WordPress/wordpress-playground/issues/692, by allowing developers to provide SQL via a resource. The queries will be run as part of the Blueprint setup process.

## What problem is it solving?

This allows a user to execute SQL in the context of a playground instance to set up the WordPress install inside.

Currently, in production, there's no easy way to set up Playground for a complex plugin like WooCommerce:

Preparing an entire custom SQLite file is tedious and may not be portable between WordPress versions
The ability to import entire SQL files was requested by multiple developers. Let's support it, or something equally expressive and convenient.

## How is the problem addressed?

The following notation is now valid for Blueprint steps:

Run a set of SQL queries:
```js
{
  step: 'runSql',
  sql: {
    resource: "literal",
    contents: "DELETE FROM wp_posts;\nDELETE FROM wp_users;",
    name: "queries.sql",
  }
}
```

## Testing Instructions

1) Checkout the project & run 

```
nx run playground-website:dev
```

Click the following link: (ensure the port number matches)
http://localhost:5400/website-server/#{%22landingPage%22:%20%22/wp-admin/edit.php%22,%20%22steps%22:[{%22step%22:%20%22login%22,%20%22username%22:%20%22admin%22,%20%22password%22:%20%22password%22},%20{%20%22step%22:%22runSql%22,%20%22sql%22:%20{%20%22resource%22:%20%22literal%22,%20%22name%22:%20%22queries.sql%22,%20%22contents%22:%20%22DELETE%20FROM%20wp_posts%22%20}%20}]}

You should see that all posts have been deleted.
